### ISSUE TITLE
Implemented M3 Typography `letterSpacing` for `bodyLarge` does not match M3 Guide Spec.

### ISSUE THREAD

**Issue by rydmike** [flutter/flutter#102121]
## Discrepancy: M3 Guide Typescale versus Flutter Typography Implementation

In this https://github.com/flutter/flutter/issues/89853#issuecomment, I made an observation that the spec in supplied [image](https://user-images.githubusercontent.com/2364772/136967954-a1e217f1-d049-4355-9bd2-5d452102bf1e.png) did not match the M3 Web [guide](https://m3.material.io/styles/typography/tokens) concerning the Typescale / Typography for TextStyle `bodyLarge` and its used `letterSpacing`.

The [image](https://user-images.githubusercontent.com/2364772/136967954-a1e217f1-d049-4355-9bd2-5d452102bf1e.png) in (https://github.com/flutter/flutter/issues/89853) has `bodyLarge` and its used `letterSpacing` defined as **0.5**
The [guide](https://m3.material.io/styles/typography/tokens) at **0.15** 

In this PR https://github.com/flutter/flutter/pull/97829/files#diff-a91306c958d84cfa7a9287e538e2c8f22bd7765d59ec1dadb2382bda832332adR751 it was implemented as **0.5**, and was merged into master as such:

https://github.com/flutter/flutter/blob/fd360c4a1d2418ad2d1ad83f7b5cc125696837f9/packages/flutter/lib/src/material/typography.dart#L749

### Question:
Is the M3 guide spec wrong?
Or is it an oversight that the correction observation did not make it into the Flutter implementation?

One or the other needs to be corrected, they can't both be right.

In [issue https://github.com/flutter/flutter/issues/89853](https://github.com/flutter/flutter/issues/89853#issuecomment-959455110) @rami-a concluded that the image was wrong and the web M3 spec correct.

Which of course means that the current Flutter implementation is incorrect.

cc: @darrenaustin @TahaTesser

**Comment by maheshmnj** [comment]
Hi @rydmike, Thanks for filing the issue. I do see 0.5 letter-spacing being used for body large on flutter master 

https://github.com/flutter/flutter/blob/fd360c4a1d2418ad2d1ad83f7b5cc125696837f9/packages/flutter/lib/src/material/typography.dart#L749

and the [M3 Typography Specs](https://m3.material.io/styles/typography/tokens) seem to indicate it should be 0.15. Labeling this issue as a bug, since this is not aligned with the Material 3 Specs

cc: @rami-a

**Comment by TahaTesser** [comment]
it makes sense looking at the M3 guide but this needs a tokens update:

https://github.com/flutter/flutter/blob/fd360c4a1d2418ad2d1ad83f7b5cc125696837f9/dev/tools/gen_defaults/data/text_style.json#L6-L8

**Comment by TahaTesser** [comment]
I wrote a quick test based on these references

```dart
  testWidgets('bodyText1 is not equal to bodyLarge', (WidgetTester tester) async {
    final ThemeData theme =  ThemeData();
    final Key bodyText1Key = UniqueKey();
    final Key bodyLargeKey = UniqueKey();
    await tester.pumpWidget(
      MaterialApp(
        home: Scaffold(
          body: Center(
            child: Text('Dash', key: bodyText1Key, style: theme.textTheme.bodyText1),
          ),
        ),
      ),
    );

    RenderParagraph paragraph = tester.renderObject(find.text('Dash')) as RenderParagraph;
    // While `text_theme.dart` references `TextStyle? get bodyText1 => bodyLarge;`.
    // `theme.textTheme.bodyLarge` paragraph size is `const Size(56.0, 14.0)`.
    expect(paragraph.size, const Size(56.0, 14.0));
      await expectLater(
        find.byKey(bodyText1Key),
        matchesGoldenFile('bodyText1.png'),
      );

    await tester.pumpWidget(
      MaterialApp(
        theme: ThemeData(useMaterial3: true),
        home: Scaffold(
          body: Center(
            child: Text('Dash', key: bodyLargeKey, style: theme.textTheme.bodyLarge),
          ),
        ),
      ),
    );
    await tester.pumpAndSettle();
    paragraph = tester.renderObject(find.text('Dash')) as RenderParagraph;
    // While `text_theme.dart` references `TextStyle? get bodyText1 => bodyLarge;`.
    // `theme.textTheme.bodyLarge` paragraph size is not equal to `bodyText1`.
    expect(paragraph.size, const Size(57.0, 20.0));
      await expectLater(
        find.byKey(bodyLargeKey),
        matchesGoldenFile('bodyLarge.png'),
      );
  });
```


https://github.com/flutter/flutter/blob/ce6188bef15c468cfd87e183b0ee14eed1a21c2d/packages/flutter/lib/src/material/text_theme.dart#L134

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/text_theme.dart#L262-L263

and recent example where `list_tile_test.dart` font size test fails when updating  `bodyText1` to `bodyLarge` in `list_tile.dart`, so i resorted to `bodyMedium`. 

https://github.com/flutter/flutter/pull/102167/files#diff-c4ef7c798c820ecda6cab38be1c2d619d5349dc478889f3a54ebbe07f60c11f2L657

**Comment by rydmike** [comment]
Maybe the token DB is correct, and thus Flutter too, but in that case the spec shown in web guide is wrong. Both can't be right, was my main point with this observation too 😄  Curious to find out which one it is supposed to be. Perhaps the token DB is correct but web never got updated.  I thought it was generated using same token db, but apparently not.

**Comment by darrenaustin** [comment]
Just heard back from the design team and it is a bug in the spec, which will be fixed in the next major roll out of the site.

The DB is correct with 0.5, so Flutter is using the right value.

**Comment by rydmike** [comment]
@darrenaustin OK thanks, good to know. I will have to update the value I have used then, as I have been using the M3 published guide spec value, but no problem, as long as it is sorted what it should be. Took a surprising long time to get an answer to simple question 😉 

I won't even be needed it all once the Typography lands in stable channel. I'm guessing at version released at Google IO.
